src/TropicalGeometry/semiring.jl

```diff
@@ -81,157 +113,158 @@ false
+ (0)
+ ∞
+ ∞
+ (0)
 ```
 """
-TropicalSemiring() = TropicalSemiring{typeof(min)}()
-TropicalSemiring(::typeof(max)) = TropicalSemiring{typeof(max)}()
-TropicalSemiring(::typeof(min)) = TropicalSemiring{typeof(min)}()
+tropical_semiring() = TropicalSemiring{typeof(min)}()
+tropical_semiring(::typeof(max)) = TropicalSemiring{typeof(max)}()
+tropical_semiring(::typeof(min)) = TropicalSemiring{typeof(min)}()
+
+
 
 ################################################################################
 #
-#  Constructors for elements
+#  Constructors for tropical numbers
 #
 ################################################################################
 
-function (R::TropicalSemiring)(u::TropicalSemiringElem)
-  @assert parent(u) === R
+function (T::TropicalSemiring)(u::TropicalSemiringElem)
+  @assert parent(u) === T
   return u
 end
 
-function (R::TropicalSemiring)(u::RingElem)
-  v = QQ(u)
-  @assert parent(v) === QQ
-  return TropicalSemiringElem(R, v)
-end
-
-function (R::TropicalSemiring)(u::Union{Integer, Rational})
-  return TropicalSemiringElem(R, QQ(u))
-end
+zero(T::TropicalSemiring) = TropicalSemiringElem(T, true)    # neutral element w.r.t. +
 
-inf(R::TropicalSemiring) = TropicalSemiringElem(R, true)
+one(T::TropicalSemiring) = TropicalSemiringElem(T, zero(QQ)) # neutral element w.r.t. *
 
-Oscar.zero(T::TropicalSemiring) = inf(T)
+inf(T::TropicalSemiring) = zero(T)
 
-Oscar.one(R::TropicalSemiring) = R(zero(QQ))
+(T::TropicalSemiring)() = zero(T)
 
-Oscar.zero(x::TropicalSemiringElem) = zero(parent(x))
 
-(R::TropicalSemiring)() = zero(R)
 
 ################################################################################
 #
-#  Basic access
+#  Conversion between tropical numbers and rational numbers.
+#  If preserve_ordering==true and minOrMax==typeof(max), flip signs
+#  (for info on tropical semiring ordering see comparison below).
 #
 ################################################################################
 
-# The underlying rational number. This is undefined for inf.
-data(x::TropicalSemiringElem) = x.data
+function (R::TropicalSemiring{typeof(min)})(x::QQFieldElem; preserve_ordering::Bool=false)
+  return TropicalSemiringElem(R,x)
+end
 
-function string(x::TropicalSemiringElem)
-    if isinf(x)
-        return "infinity"
-    end
-    return string(data(x))
+function (R::TropicalSemiring{typeof(max)})(x::QQFieldElem; preserve_ordering::Bool=false)
+  return (preserve_ordering ? TropicalSemiringElem(R,-x) : TropicalSemiringElem(R,x))
 end
 
-# Test if something is inf.
-isinf(x::TropicalSemiringElem) = x.isinf
+function (R::TropicalSemiring)(x::Union{Integer, Rational}; preserve_ordering::Bool=false)
+  return R(QQ(x),preserve_ordering=preserve_ordering)
+end
 
-Oscar.parent(x::TropicalSemiringElem{T}) where T = TropicalSemiring{T}()
+function (R::TropicalSemiring)(x::RingElem; preserve_ordering::Bool=false)
+  x = QQ(x)
+  @assert parent(x)==QQ "cannot convert object of type $(repr(typeof(x)))"
```

Comment:
```suggestion
  @req parent(x)==QQ "cannot convert object of type $(repr(typeof(x)))"
```
Please use `@req` instead of `@assert`.